src/main/java/de/tum/in/www1/artemis/service/notifications/NotificationSettingsService.java

```diff
@@ -121,22 +145,17 @@ public NotificationSettingsService(NotificationSettingRepository notificationSet
     public boolean checkIfNotificationOrEmailIsAllowedBySettingsForGivenUser(Notification notification, User user, NotificationSettingsCommunicationChannel communicationChannel) {
         NotificationType type = findCorrespondingNotificationType(notification.getTitle());
 
-        Set<NotificationSetting> notificationSettings = notificationSettingRepository.findAllNotificationSettingsForRecipientWithId(user.getId());
-
-        Set<NotificationType> deactivatedTypes;
+        Set<NotificationSetting> decidedNotificationSettings = notificationSettingRepository.findAllNotificationSettingsForRecipientWithId(user.getId());
+        Set<NotificationSetting> notificationSettings = new HashSet<>(decidedNotificationSettings);
 
-        // the urgent emails were already sent at this point
-        // if the user has not yet changed his settings they will be of size 0 -> use default
-        if (notificationSettings.isEmpty()) {
-            deactivatedTypes = findDeactivatedNotificationTypes(communicationChannel, DEFAULT_NOTIFICATION_SETTINGS);
-        }
-        else {
-            deactivatedTypes = findDeactivatedNotificationTypes(communicationChannel, notificationSettings);
-        }
-
-        if (deactivatedTypes.isEmpty()) {
-            return true;
+        // for those notification types that are not explicitly set by the user, we use the default settings
+        Set<String> decidedIds = decidedNotificationSettings.stream().map(NotificationSetting::getSettingId).collect(Collectors.toSet());
```

Comment:
This fixes a bug that I noticed. When a new notification setting has been introduced, there is no corresponding row in the notification settings table. This caused e-mail to always be sent even though the default settings actually specify no email.